install/kubernetes/istio-sidecar-initializer.yaml

```diff
@@ -0,0 +1,79 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: istio-inject
+  namespace: istio-system
+data:
+  config: |-
+    policy: "enabled"
+    namespaces: [""] # everything, aka v1.NamepsaceAll, aka cluster-wide
+    initializerName: "sidecar.initializer.istio.io"
+    params:
+      initImage: gcr.io/istio-testing/proxy_init:5acaa9e26e7d9b239cbc72da63f5f9a3dac5a54e
+      proxyImage: gcr.io/istio-testing/proxy:5acaa9e26e7d9b239cbc72da63f5f9a3dac5a54e
+      verbosity: 2
+      version: 5acaa9e26e7d9b239cbc72da63f5f9a3dac5a54e
+      meshConfigMapName: istio
+      imagePullPolicy: IfNotPresent
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-sidecar-initializer-service-account
+  namespace: istio-system
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: istio-sidecar-initializer
+  namespace: istio-system
+  annotations:
+    sidecar.istio.io/inject: "false"
+  initializers:
+    pending: []
+  labels:
+    istio: istio-sidecar-initializer
+spec:
+  replicas: 1
+  template:
+    metadata:
+      name: istio-sidecar-initializer
+      labels:
+        istio: sidecar-initializer
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: istio-sidecar-initializer-service-account
+      containers:
+        - name: sidecar-initializer
+          image: gcr.io/istio-testing/sidecar_initializer:5acaa9e26e7d9b239cbc72da63f5f9a3dac5a54e
+          imagePullPolicy: IfNotPresent
+          args:
+            - --port=8083
+            - --namespace=istio-system
+            - -v=2
+          volumeMounts:
+          - name: config-volume
+            mountPath: /etc/istio/config
+      volumes:
+      - name: config-volume
+        configMap:
+          name: istio
+---
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: InitializerConfiguration
+metadata:
+  name: istio-sidecar
+initializers:
+  - name: sidecar.initializer.istio.io
+    rules:
+      - apiGroups:
+          - "*"
+        apiVersions:
+          - "*"
+        resources:
+          - deployments
+          - statefulsets
+          - jobs
+          - daemonsets
```

Comment:
Do we need other workloads?

Reply:
* Initializer doesn't support cronjob yet. 
* replicatsets and replicationcontrollers had some issues with current initializer support that aren't resolved yet. Deployments are recommended instead of using replicasets or replicationctonrollers (see [here](https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller/#alternatives-to-replicationcontroller))
* pod also did not work but isn't recommended on its own since some controllers will complain if the pod template spec in the deployment doesn't match the pod spec itself.